java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -38,37 +38,51 @@
 public class RangeEqualsVisitor implements VectorVisitor<Boolean, Range> {
   private ValueVector left;
   private ValueVector right;
-  private boolean isTypeCheckNeeded;
+
+  private TypeEqualsVisitor typeVisitor;
+  private boolean checkType;
   private boolean typeCompareResult;
 
+  /**
+   * Constructs a new instance with default type visitor (do not check name and metadata).
+   */
+  public RangeEqualsVisitor(ValueVector left, ValueVector right, boolean checkType) {
+    this (left, right, checkType, new TypeEqualsVisitor(right));
+  }
+
   /**
    * Constructs a new instance.
    *
    * @param left left vector
    * @param right right vector
-   * @param isTypeCheckNeeded type check needed
+   * @param checkType whether need check type
+   * @param typeVisitor type visitor
    */
-  public RangeEqualsVisitor(ValueVector left, ValueVector right, boolean isTypeCheckNeeded) {
+  public RangeEqualsVisitor(
+      ValueVector left,
+      ValueVector right,
+      boolean checkType,
```

Comment:
it seems redundant to both "true" and a typeVisitor when type checking is desired.  I think we should maybe eliminate checkType from her and use nullness of typeVisitor to determine when to perform the check.

Reply:
right, revised.